### ISSUE TITLE
Experiment Results

### ISSUE THREAD

**Issue by StephenEkaputra** [SHI-Labs/Compact-Transformers#3]
Hi, Ali Hassani
Thank you for sharing your amazing work!

If feasible, I would like to ask a question.
I wonder what training settings you use for CIFAR100?
Because when I use the default setting with resnet18, my best result is 58% differ from the paper which is 63.41%.
I have tried it twice and tweak the warm up to 5 or 10 epochs.
In addition, CCT-2 result is 65.74% in the paper is 66.90%.

Resnet18 pytorch result:
Accu Epoch 190 | 57.7
Accu Epoch 191 | 57.85
Accu Epoch 192 | 57.87
Accu Epoch 193 | 57.92
Accu Epoch 194 | 57.61
Accu Epoch 195 | 57.97
Accu Epoch 196 | 57.78
Accu Epoch 197 | 57.92
Accu Epoch 198 | 58.02
Accu Epoch 199 | 57.94
Accu Epoch 200 | 57.84

Thanks

**Comment by stevenwalton** [comment]
Hey Stephen, 

Just a note that our results are the best out of 4 runs (Table 1). So 65.74 is a bit low but not too surprising. 

*ResNet 18*
- Learning rate: 0.1
- Cosine learning rate
- Weight decay: 1e-4
- 200 epochs
- warmup: 10
- batch size: 128
- SGD optimizer
- label smoothing: 0.1
- auto augment 

I'm willing to bet that the mistake is not using the autoaugment. This had big changes in performance for us. We use a [modified version](https://github.com/SHI-Labs/Compact-Transformers/blob/main/utils/autoaug.py) or [AutoAugment](https://github.com/DeepVoltaire/AutoAugment/blob/master/autoaugment.py).

*CCT-2*
- Learning rate: 5e-4
- Cosine Learning rate
- Weight decay 3e-2
- 200 epochs
- warmup: 10
- Batch size 128
- Adamw optimizer
- label smoothing 0.1
- auto augment

I believe all these are the default options (we didn't expose everything to the public repo and are still in the process of doing so. We'll have the final version out once the conference submission comes up). I should also note that we reported our best runs. Our worst run in that set is 65.24% and we have an average of 65.87%. Your result seems perfectly in range.

**Comment by StephenEkaputra** [comment]
Thank you Steven for your great answer.
It is very enlighten me.
If I may ask again, I wonder why the learning rate and weight decay of resnet18 and cct2 are different? Why don't we use the same Lr and weight decay to all models?

Thank you in advance.

**Comment by stevenwalton** [comment]
No problem, happy to help. All hyper-parameters, learning rate included, are going to be dependent upon the model. These are just what we found to work best (note, they might not actually even be the best!). So we did a search of different parameters and these were the best we found.

**Comment by stevenwalton** [comment]
I'm closing the issue now, if you still have questions feel free to open it back up.
